modules/command-engine/core/src/test/scala/surge/internal/domain/SurgeMessagePipelineSpec.scala

```diff
@@ -77,277 +64,240 @@ class SurgeMessagePipelineSpec
+    with Matchers
+    with SurgeMessagePipelineSpecLike {
 
-          val result = restarted.futureValue
-          result shouldEqual Ack()
-        }
-      }
-    }
+  implicit override val patienceConfig: PatienceConfig =
+    PatienceConfig(timeout = scaled(Span(15, Seconds)), interval = scaled(Span(50, Milliseconds)))
 
-    "subscribe to health.signals via stream on start" in {
-      withRunningKafkaOnFoundPort(config) { _ =>
-        withTestContext { ctx =>
-          import ctx._
-          createCustomTopic(businessLogic.kafka.eventsTopic.name, Map.empty)
-          createCustomTopic(businessLogic.kafka.stateTopic.name, Map.empty)
-
-          // Stream should be a subscriber
-          signalStreamProvider
-            .bus()
-            .subscriberInfo()
-            .exists(s => s.name == "surge.internal.health.windows.stream.sliding.SlidingHealthSignalStreamImpl") shouldEqual true
-        }
-      }
-    }
+  override val defaultConfig: Config = ConfigFactory.load()
+  implicit val config: EmbeddedKafkaConfig = EmbeddedKafkaConfig(kafkaPort = defaultConfig.getInt("kafka.port"))
 
-    "Register all wrapped components with the HealthSignalBus" in {
-      withRunningKafkaOnFoundPort(config) { _ =>
-        withTestContext { ctx =>
-          import ctx._
+  override val actorSystem: ActorSystem = system
 
-          createCustomTopic(businessLogic.kafka.eventsTopic.name, Map.empty)
-          createCustomTopic(businessLogic.kafka.stateTopic.name, Map.empty)
+  override def beforeAll(): Unit = {
+    EmbeddedKafka.start()
+    createCustomTopic(
+      topic = businessLogic.kafka.stateTopic.name,
+      partitions = 5,
+      topicConfig = Map(TopicConfig.CLEANUP_POLICY_CONFIG -> TopicConfig.CLEANUP_POLICY_COMPACT))
+    createCustomTopic(topic = businessLogic.kafka.eventsTopic.name, partitions = 5)
+    super.beforeAll()
+  }
 
-          val bus = pipeline.signalBus
-          // Retrieve Registrations and verify SurgeMessagePipeline is registered
-          eventually {
-            val registrations = bus.registrations().futureValue
-            val registrationNames = registrations.map(_.componentName)
-            registrationNames should contain("surge-message-pipeline")
+  override def afterAll(): Unit = {
+    EmbeddedKafka.stop()
+    // FIXME verifySystemShutdown should be true, but this does not shut down in a reasonable amount of time
+    TestKit.shutdownActorSystem(system, duration = 30.seconds, verifySystemShutdown = false)
+    super.afterAll()
+  }
 
-            registrationNames should contain("state-store-kafka-streams")
-            val kStreamsRegistration = registrations.find(r => r.componentName == "state-store-kafka-streams").get
-            kStreamsRegistration.restartSignalPatterns.map(p => p.pattern()) should contain("kafka.streams.fatal.error")
+  val ctx: TestContext = createTestContext()
+  val pipeline: SurgeMessagePipeline[State, BaseTestCommand, BaseTestEvent] = ctx.pipeline
```

Comment:
The class initializes a single SurgeMessagePipeline. 
Because we can't start and stop the pipeline in a fast manner between the tests and it causes Actor name is not unique issue.